### PR TITLE
implement clock mechanics

### PR DESCRIPTION
**Pull request by Koliras**
[DEMO LINK](https://Koliras.github.io/react_clock/)

### CODE REVIEW

**Review by FrontKid — APPROVED** [review]
Good  job!

File: src/components/Clock/Clock.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  name: string,
+};
+
+type State = {
+  today: Date,
+};
+
+export class Clock extends React.Component<Props> {
+  state: State = {
+    today: new Date(),
```
Comment (roman-mirzoian) [inline src/components/Clock/Clock.tsx:13]:
```suggestion
    currentDate: new Date(),
```

File: src/components/Clock/Clock.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {this.props.name}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {this.state.today.toUTCString().slice(-12, -4)}
```
Comment (roman-mirzoian) [inline src/components/Clock/Clock.tsx:40]: You make similar calculations twice, so you should make a separate helper function for this.

**Review by roman-mirzoian — CHANGES_REQUESTED** [review]
Good job, but there is quite a bit of work to be done.

**Pull request closed without merging**